keps/sig-node/4742-node-topology-downward-api/README.md

```diff
@@ -0,0 +1,417 @@
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [X] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [X] (R) KEP approvers have approved the KEP status as `implementable`
+- [X] (R) Design details are appropriately documented
+- [X] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [X] e2e Tests for all Beta API Operations (endpoints)
+  - [X] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [X] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [X] (R) Graduation criteria is in place
+  - [X] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [X] (R) Production readiness review completed
+- [X] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Many workloads benefit from knowing the underlying topology that a Pod is scheduled in.
+However, today there is no mechanism in Kubernetes to pass down node topology information to Pods.
+The KEP outlines how node topology labels will made available to Pods via the downward API.
+
+Exposing node topology labels through the downward API provides a native method
+from which to attain node label values in pods without the need of a sidecar
+to extract information.
+
+## Motivation
+
+We’d like to change the runtime behavior of containers based on node labels.
+In our case, we’re using a CNI with DaemonSets to perform network setup, and
+would like to configure the network differently based on the presence of a node
+label.
```

Comment:
Who's “our“? Kubernetes project, or one end user?

Reply:
To be fair - @andrewsykim took over the KEP from https://github.com/kubernetes/enhancements/pull/4747

Reply:
Maybe it's come over as harsh, but I'm keen we don't merge KEPs - which are kind of official documents - where there's ambiguity that could confuse someone. Maybe this is the first KEP they read and it leaves the wrong impression about project-wide opinions.

Reply:
oh, it's fair comment, just clarifying history

Reply:
I ported the KEP over and just filled in the PRR parts to avoid missing the PRR freeze :) I will update the other sections soon